src/analyzer/vamp/vamppluginloader.cpp

```diff
@@ -1,58 +1,137 @@
-/*
- * vamppluginloader.cpp
- *
- *  Created on: 23/jan/2012
- *      Author: Tobias Rafreider
- *
- * This is a thread-safe wrapper class around Vamp's
- * PluginLoader class.
- */
 #include "analyzer/vamp/vamppluginloader.h"
 
-using Vamp::Plugin;
-using Vamp::PluginHostAdapter;
+#include <mutex>
 
-VampPluginLoader* VampPluginLoader::s_instance = NULL;
-QMutex VampPluginLoader::s_mutex;
+#include <QCoreApplication>
+#include <QDesktopServices>
+#include <QDir>
+#include <QStringList>
+#include <QStringBuilder>
 
-VampPluginLoader::VampPluginLoader() {
-    m_pVampPluginLoader = Vamp::HostExt::PluginLoader::getInstance();
+#include "util/logger.h"
+
+#ifdef __WINDOWS__
+    #include <windows.h>
+    #define PATH_SEPARATOR ";"
+#else
+    #define PATH_SEPARATOR ":"
+#endif
+
+
+namespace mixxx {
+
+namespace {
+
+Logger kLogger("VampPluginLoader");
+
+Vamp::HostExt::PluginLoader* pPluginLoader = nullptr;
```

Comment:
I have stumbled over this, because it is hard to distinguish from a local. Can't we use just instance()?
Isn't a name space global the same as a static variable? So we could use the s prefix?